warehouse/query-core/src/main/java/datawave/query/planner/DefaultQueryPlanner.java

```diff
@@ -1166,108 +1173,141 @@ protected ASTJexlScript processTree(final ASTJexlScript originalQueryTree, Shard
             stopwatch.stop();
         }
         
-        if (!disableBoundedLookup) {
-            stopwatch = timers.newStartedStopwatch("DefaultQueryPlanner - Expand bounded query ranges");
-            
-            // Expand any bounded ranges into a conjunction of discrete terms
+        ParallelIndexExpansion regexExpansion = null;
+        if (!disableBoundedLookup && (planMetadata.hasRegexNodes() || planMetadata.hasAnyFieldNodes())) {
             try {
-                ParallelIndexExpansion regexExpansion = new ParallelIndexExpansion(config, scannerFactory, metadataHelper, expansionFields,
-                                config.isExpandFields(), config.isExpandValues(), config.isExpandUnfieldedNegations());
+                stopwatch = timers.newStartedStopwatch("DefaultQueryPlanner - ParallelIndexExpansion expand regex terms");
+                regexExpansion = new ParallelIndexExpansion(config, scannerFactory, metadataHelper, expansionFields, config.isExpandFields(),
+                                config.isExpandValues(), config.isExpandUnfieldedNegations());
                 queryTree = (ASTJexlScript) regexExpansion.visit(queryTree, null);
-                if (log.isDebugEnabled()) {
+                
+            } catch (InstantiationException | IllegalAccessException | TableNotFoundException e) {
+                QueryException qe = new QueryException(DatawaveErrorCode.METADATA_ACCESS_ERROR, e);
+                throw new DatawaveFatalQueryException(qe);
+            } catch (CannotExpandUnfieldedTermFatalException e) {
+                if (null != e.getCause() && e.getCause() instanceof DoNotPerformOptimizedQueryException)
+                    throw (DoNotPerformOptimizedQueryException) e.getCause();
+                QueryException qe = new QueryException(DatawaveErrorCode.INDETERMINATE_INDEX_STATUS, e);
+                throw new DatawaveFatalQueryException(qe);
+            } finally {
+                if (log.isDebugEnabled())
                     logQuery(queryTree, "Query after expanding regex:");
-                }
+                stopwatch.stop();
+            }
+        }
+        
+        if (!disableBoundedLookup && planMetadata.boundedRangesPossible()) {
+            try {
+                stopwatch = timers.newStartedStopwatch("DefaultQueryPlanner - Expand bounded query ranges");
                 queryTree = RangeConjunctionRebuildingVisitor.expandRanges(config, scannerFactory, metadataHelper, queryTree, config.isExpandFields(),
                                 config.isExpandValues());
-                if (log.isDebugEnabled()) {
+            } catch (TableNotFoundException | ExecutionException e) {
```

Comment:
The range conjunction rebuilding visitor may end up removing bounded ranges if they expanded to the empty set making the planmetadata invalid yet again.